x-pack/solutions/observability/plugins/apm/public/components/fleet_integration/apm_policy_form/settings_form/index.tsx

```diff
@@ -177,3 +191,18 @@ function AdvancedOptions({ children }: { children: React.ReactNode }) {
     </>
   );
 }
+
+function getRegistryEntryVar(key: string, packageInfo: PackageInfo | undefined) {
+  if (
+    packageInfo &&
+    packageInfo.policy_templates &&
+    packageInfo.policy_templates.length > 0 &&
+    'inputs' in packageInfo.policy_templates[0] &&
+    Array.isArray(packageInfo.policy_templates[0].inputs) &&
+    packageInfo.policy_templates[0].inputs.length > 0 &&
+    packageInfo.policy_templates[0].inputs[0].vars
+  ) {
```

Comment:
Would this work instead?

```suggestion
  if (
    packageInfo?.policy_templates?.[0]?.inputs?.[0]?.vars
  ) {
```

Reply:
it works per se if I shorten it but the linter is not happy. This looks a bit funky and I don't like it either, but the alternative is 
```
  if (
    packageInfo &&
    packageInfo.policy_templates &&
    packageInfo.policy_templates.length > 0 &&
    packageInfo.policy_templates[0].inputs.length > 0 &&
    packageInfo.policy_templates[0].inputs[0].vars
  )
```

and some `ts-ignore` lines here and there. The reason is that `inputs` exists only in one of the types of this union type:
https://github.com/criamico/kibana/blob/a60bde8d65ce41f14aced142fb273457ef928e30/x-pack/platform/plugins/shared/fleet/common/types/models/epm.ts#L273-L275

so I had to add the type guard as well

Reply:
Maybe we could break the condition? Just brainstorming ideas to try to simplify the condition. 

```ts
const { policy_templates = [] } = packageInfo || {};
const policyTemplate = policy_templates[0];
const inputs = 'inputs' in policy_templates[0] ? policy_templates[0].inputs : undefined;

if (inputs && Array.isArray(inputs) && inputs.length > 0 && inputs[0].vars) {
}
```

I don't want to block the PR because of this though.